azure-pipelines.yml

```diff
@@ -1,342 +1,416 @@
+                  environment: ${{ component.env }}
+                  component: managed-identity
+                  location: ${{ parameters.location }}
+                  serviceConnection: ${{ component.serviceConnection }}
+                  terraformInitSubscription: ${{ variables.tfInitSub }}
+                  product: ${{ variables.product }}
+                  tfVarsFile: "$(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/environments/managed-identity/${{ component.env }}.tfvars"
+                  initCommandOptions: >
+                    -var subscription_id=${{ component.terraformSubscriptionID }}
+                    -backend-config resource_group_name=${{ component.backendAzureRmResourceGroupName }}
+                  planCommandOptions: >
+                    -var subscription_id=${{ component.terraformSubscriptionID }}
+                    -var control_vault=$(controlKeyVault)
+                    -var project=$(project)
 
-  - stage: Managed_Identity
-    displayName: 'Managed_Identity'
-    dependsOn: Network
-    jobs:
-      - job: Managed_Identity
-        steps:
-          - template: steps/terraform.yaml@cnp-azuredevops-libraries
-            parameters:
-              overrideAction: ${{ parameters.action }}
-              environment: ${{ parameters.env }}
-              component: managed-identity
-              location: ${{ parameters.location }}  
-              serviceConnection: $(serviceConnection)
-              terraformInitSubscription: ${{ variables.tfInitSub }}
-              product: ${{ variables.product }}
-              tfVarsFile: "$(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/environments/managed-identity/${{ parameters.env }}.tfvars"
-              initCommandOptions: >
-                -var subscription_id=${{ variables.terraformSubscriptionID }}
-                -backend-config resource_group_name=$(backendAzureRmResourceGroupName)
-              planCommandOptions: >
-                -var subscription_id=${{ variables.terraformSubscriptionID }}
-                -var control_vault=$(controlKeyVault)
-                -var project=$(project)
+      - stage: "Aks_${{ component.env }}"
+        displayName: "${{ upper(component.env) }}: AKS clusters"
+        dependsOn:
+          - "CheckingClusters_${{ component.env }}"
+          - "Managed_Identity_${{ component.env }}"
+        variables:
+          existing_clusters: $[stageDependencies.CheckingClusters_${{ component.env }}.CheckingClusters.outputs['check_clusters.cluster_deploy']]
+          deploy_cluster: ${{ parameters.cluster }}
+        condition: |
+            or(
+              and(
+                eq(variables['Build.Reason'], 'Manual'),
+                or(
+                  and(
+                    eq('${{ parameters.cluster }}', 'All'),
+                    in(dependencies.CheckingClusters_${{ component.env }}.outputs['CheckingClusters.check_clusters.cluster_deploy'], '00', '01')
```

Comment:
Won't this  cause it to constantly skip this bit because the condition says, if parameters.cluster is set to All, **and** the current cluster is either 00 or 01? So it will try to check the or condition every time right?


I still think there's some confusion in this and that it can be simplified to something like 
```
             and(
              eq(variables['Build.Reason'], 'Manual'),
              eq(dependencies.CheckingClusters_${{ component.env }}.outputs['CheckingClusters.check_clusters.cluster_deploy'], '${{ parameters.cluster }}')
            )
```

If this was the total condition for this stage, wouldn't that meet the logic you want?